packages/php-wasm/compile/build-assets/php_wasm.c

```diff
@@ -802,10 +815,38 @@ static void wasm_sapi_register_server_variables(zval *track_vars_array TSRMLS_DC
 	if (value != NULL) {
 		php_register_variable("SCRIPT_NAME", value, track_vars_array TSRMLS_CC);
 		php_register_variable("SCRIPT_FILENAME", value, track_vars_array TSRMLS_CC);
-		php_register_variable("PHP_SELF", value, track_vars_array TSRMLS_CC);
 		php_register_variable("REQUEST_URI", value, track_vars_array TSRMLS_CC);
 	}
 
+	int php_self_set = 0;
+	if (wasm_server_context->document_root != NULL && wasm_server_context->path_translated != NULL)
+	{
+		/**
+		 * PHP_SELF is the script path relative to the document rooth.
+		 *
+		 * For example:
+		 * 
+		 * If the document root is /var/www/html and the requested path is /dir/index.php,
+		 * PHP_SELF will be /dir/index.php.
+		 * 
+		 * If the document root is /var/www/html and the requested path /nice/urls is
+		 * served from /var/www/html/another/directory/script.php, PHP_SELF will be
+		 * /another/directory/script.php
+		 */
+		// Confirm path translated starts with the document root
+		if (strncmp(wasm_server_context->document_root, wasm_server_context->path_translated, strlen(wasm_server_context->document_root)) == 0) {
+			// Substring of path translated starting after document root
+			char *php_self = wasm_server_context->path_translated + strlen(wasm_server_context->document_root);
+			php_register_variable("PHP_SELF", strdup(php_self), track_vars_array TSRMLS_CC);
+			php_self_set = 1;
+		}
+	}
+
+	if(php_self_set == 0) {
```

Comment:
Oops, need to check for null values here:

```suggestion
	if(php_self_set == 0 && value != null) {
```

Reply:
added here https://github.com/WordPress/wordpress-playground/pull/280/commits/12c8eb26daa3a2ea69effc35e20dcf908d90c03c